From: coolsnowwolf <coolsnowwolf@gmail.com>
Subject: [PATCH] ath10k-ct: use strscpy for coredump strings

Match the in-kernel ath10k driver and avoid fortify false positives when
copying strings into the dynamically allocated crash dump structure.

Signed-off-by: coolsnowwolf <coolsnowwolf@gmail.com>
---
 ath10k-5.15/coredump.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/ath10k-5.15/coredump.c
+++ b/ath10k-5.15/coredump.c
@@ -1545,7 +1545,7 @@
 	mutex_lock(&ar->dump_mutex);
 
 	dump_data = (struct ath10k_dump_file_data *)(buf);
-	strlcpy(dump_data->df_magic, "ATH10K-FW-DUMP",
+	strscpy(dump_data->df_magic, "ATH10K-FW-DUMP",
 		sizeof(dump_data->df_magic));
 	dump_data->len = cpu_to_le32(len);
 
@@ -1572,11 +1572,11 @@
 	dump_data->ram_bss_addr =
 		cpu_to_le32(ar->running_fw->fw_file.ram_bss_addr);
 
-	strlcpy(dump_data->fw_ver, ar->hw->wiphy->fw_version,
+	strscpy(dump_data->fw_ver, ar->hw->wiphy->fw_version,
 		sizeof(dump_data->fw_ver));
 
 	dump_data->kernel_ver_code = 0;
-	strlcpy(dump_data->kernel_ver, init_utsname()->release,
+	strscpy(dump_data->kernel_ver, init_utsname()->release,
 		sizeof(dump_data->kernel_ver));
 
 	dump_data->tv_sec = cpu_to_le64(crash_data->timestamp.tv_sec);
